datasetinsights/evaluation_metrics/average_precision_2d.py

```diff
@@ -196,3 +185,105 @@ def n_point_interpolated_ap(recall, precision, point=11):
                 cc.append(p)
 
         return ap
+
+
+class AveragePrecisionIOU50(EvaluationMetric):
+    """2D Bounding Box Average Precision at IOU = 50%.
+
+    This implementation would calculate AP@50IOU for each label.
+
+    Attributes:
```

Comment:
[nit] The attributes ap here seems to suggests `AveragePrecisionIOU50.ap` and `MeanAveragePrecisionIOU50` store the same info. They are not (I think). Maybe use slightly different name might be better. 

Also, should this a public instance variable?

Reply:
They both use `AveragePrecision` to calculate Average Precision at IOU = 50%. The only difference is in the `compute()` method. 
If they are not public instance variables, do you want to make them as class attributes?

Reply:
I don't think `__getattr__` would work. Going to stick with the previous version.